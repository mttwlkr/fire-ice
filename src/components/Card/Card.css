.Card ul {
  list-style-type: none;
}