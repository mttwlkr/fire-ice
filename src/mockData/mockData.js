export const mockHousesArray = [
  { 
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    currentLord: "https://www.anapioficeandfire.com/api/characters/681",
    diedOut: "",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words: "",
  },
  { 
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    currentLord: "https://www.anapioficeandfire.com/api/characters/681",
    diedOut: "",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words: "",
  }  
]

const mockHousesAction = {houses: [
    { 
      ancestralWeapons: ["Lady Forlorn"],
      cadetBranches: [],
      coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
      currentLord: "https://www.anapioficeandfire.com/api/characters/681",
      diedOut: "",
      founded: "",
      founder: "https://www.anapioficeandfire.com/api/characters/256",
      heir: "https://www.anapioficeandfire.com/api/characters/677",
      name: "House Corbray of Heart's Home",
      overlord: "https://www.anapioficeandfire.com/api/houses/7",
      region: "",
      seats: ["Heart's Home"],
      swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
      titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
      url: "https://www.anapioficeandfire.com/api/houses/81",
      words: "",
    },
    { 
      ancestralWeapons: ["Lady Forlorn"],
      cadetBranches: [],
      coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
      currentLord: "https://www.anapioficeandfire.com/api/characters/681",
      diedOut: "",
      founded: "",
      founder: "https://www.anapioficeandfire.com/api/characters/256",
      heir: "https://www.anapioficeandfire.com/api/characters/677",
      name: "House Corbray of Heart's Home",
      overlord: "https://www.anapioficeandfire.com/api/houses/7",
      region: "",
      seats: ["Heart's Home"],
      swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
      titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
      url: "https://www.anapioficeandfire.com/api/houses/81",
      words: "",
    }  
  ]
}

export const mockCardWithSwornMembers = { 
  ancestralWeapons: ["Lady Forlorn"],
  cadetBranches: [],
  coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
  currentLord: "https://www.anapioficeandfire.com/api/characters/681",
  diedOut: "",
  founded: "",
  founder: "https://www.anapioficeandfire.com/api/characters/256",
  heir: "https://www.anapioficeandfire.com/api/characters/677",
  name: "House Corbray of Heart's Home",
  overlord: "https://www.anapioficeandfire.com/api/houses/7",
  region: "",
  seats: ["Heart's Home"],
  swornMembers: [
    {url: "https://www.anapioficeandfire.com/api/characters/255", name: "Corwyn Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/256", name: "Corwyn Corbray", gender: "Male", culture: "Andal", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/447", name: "Gwayne Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/654", name: "Lucas Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/677", name: "Lyn Corbray", gender: "Male", culture: "", born: "At Heart's Home"},
    {url: "https://www.anapioficeandfire.com/api/characters/681", name: "Lyonel Corbray", gender: "Male", culture: "", born: "In 260 AC (roughly), at Heart's Home"},
    {url: "https://www.anapioficeandfire.com/api/characters/871", name: "Rhaena Targaryen", gender: "Female", culture: "", born: "In 116 AC, at Pentos"},
    {url: "https://www.anapioficeandfire.com/api/characters/1182", name: "Baelish", gender: "Male", culture: "Braavosi", born: ""}
  ],
  titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
  url: "https://www.anapioficeandfire.com/api/houses/81",
  words: "",
}


