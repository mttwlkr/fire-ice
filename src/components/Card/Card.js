import React, { Component } from 'react';
import './Card.css';

class Card extends Component {
  constructor() {
    super();
    this.state = {
      clicked: false
    }
  }

  handleClick = () => {
    this.setState({ clicked: !this.state.clicked })
  }

  render() {
    const { house } = this.props

    const members = house.swornMembers.map(member => {
      return <li>{member.name}</li>
    })

    const displayMembers = this.state.clicked ? members : '' 

    return (
      <div 
        className='Card'
        onClick={this.handleClick}  
      >

      { house &&
        <div>
          <h1>{house.name}</h1>
          <p>{house.founded}</p> 
          <p>{house.seats}</p>
          <p>{house.titles}</p>
          <p>{house.coatOfArms}</p>
          <p>{house.ancestralWeapons}</p>
          <p>{house.words}</p>
          <ul>{displayMembers}</ul>
        </div>
      }
      </div>
    )  
  }
}

export default Card
