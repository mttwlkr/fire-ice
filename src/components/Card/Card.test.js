import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js'
import { mockCardWithSwornMembers } from '../../mockData/mockData.js'

describe('Card', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card 
      house={mockCardWithSwornMembers}
    />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update the state when clicked', () => {
    wrapper.find('.Card').simulate('click')
    expect(wrapper.state('clicked')).toEqual(true)
  })
})