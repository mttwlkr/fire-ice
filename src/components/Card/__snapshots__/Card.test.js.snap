// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "cadetBranches": Array [],
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
            "diedOut": "",
            "founded": "",
            "founder": "https://www.anapioficeandfire.com/api/characters/256",
            "heir": "https://www.anapioficeandfire.com/api/characters/677",
            "name": "House Corbray of Heart's Home",
            "overlord": "https://www.anapioficeandfire.com/api/houses/7",
            "region": "",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": Array [
              Object {
                "born": "",
                "culture": "",
                "gender": "Male",
                "name": "Corwyn Corbray",
                "url": "https://www.anapioficeandfire.com/api/characters/255",
              },
              Object {
                "born": "",
                "culture": "Andal",
                "gender": "Male",
                "name": "Corwyn Corbray",
                "url": "https://www.anapioficeandfire.com/api/characters/256",
              },
              Object {
                "born": "",
                "culture": "",
                "gender": "Male",
                "name": "Gwayne Corbray",
                "url": "https://www.anapioficeandfire.com/api/characters/447",
              },
              Object {
                "born": "",
                "culture": "",
                "gender": "Male",
                "name": "Lucas Corbray",
                "url": "https://www.anapioficeandfire.com/api/characters/654",
              },
              Object {
                "born": "At Heart's Home",
                "culture": "",
                "gender": "Male",
                "name": "Lyn Corbray",
                "url": "https://www.anapioficeandfire.com/api/characters/677",
              },
              Object {
                "born": "In 260 AC (roughly), at Heart's Home",
                "culture": "",
                "gender": "Male",
                "name": "Lyonel Corbray",
                "url": "https://www.anapioficeandfire.com/api/characters/681",
              },
              Object {
                "born": "In 116 AC, at Pentos",
                "culture": "",
                "gender": "Female",
                "name": "Rhaena Targaryen",
                "url": "https://www.anapioficeandfire.com/api/characters/871",
              },
              Object {
                "born": "",
                "culture": "Braavosi",
                "gender": "Male",
                "name": "Baelish",
                "url": "https://www.anapioficeandfire.com/api/characters/1182",
              },
            ],
            "titles": undefined,
            "url": "https://www.anapioficeandfire.com/api/houses/81",
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
                House Corbray of Heart's Home
        </h1>
        <p />
        <p>
                Heart's Home
        </p>
        <p />
        <p>
                Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
        </p>
        <p>
                Lady Forlorn
        </p>
        <p />
        <ul />
</div>,
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            House Corbray of Heart's Home
</h1>,
          <p />,
          <p>
            Heart's Home
</p>,
          <p />,
          <p>
            Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</p>,
          <p>
            Lady Forlorn
</p>,
          <p />,
          <ul />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Heart's Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Heart's Home",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          },
          "ref": null,
          "rendered": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Lady Forlorn",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Lady Forlorn",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
                    House Corbray of Heart's Home
          </h1>
          <p />
          <p>
                    Heart's Home
          </p>
          <p />
          <p>
                    Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
          </p>
          <p>
                    Lady Forlorn
          </p>
          <p />
          <ul />
</div>,
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              House Corbray of Heart's Home
</h1>,
            <p />,
            <p>
              Heart's Home
</p>,
            <p />,
            <p>
              Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</p>,
            <p>
              Lady Forlorn
</p>,
            <p />,
            <ul />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "House Corbray of Heart's Home",
            },
            "ref": null,
            "rendered": "House Corbray of Heart's Home",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Heart's Home",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Heart's Home",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            },
            "ref": null,
            "rendered": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Lady Forlorn",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Lady Forlorn",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
